pkg/gadgets/trace/open/tracer/tracer.go

```diff
@@ -101,6 +101,10 @@ func (t *Tracer) install() error {
 	consts := make(map[string]interface{})
 	consts["get_full_path"] = t.config.FullPath
 
+	if t.config.FullPath {
+		spec.Maps["start"].MaxEntries = 1024
+	}
+
```

Comment:
I don't think it'll work as the structure always has `full_fname` defined:

https://github.com/inspektor-gadget/inspektor-gadget/blob/5438f5e9e0c9f2cbe781a1eb288f7765a7634806/pkg/gadgets/trace/open/tracer/bpf/opensnoop.h#L10C1-L24C3

I think an `#ifdef` logic is needed here. An  ["Array of Length Zero"](https://gcc.gnu.org/onlinedocs/gcc/Zero-Length.html) could be a more elegant solution, but I don't know if clang and the verifier will be happy with it.